sui/gas-service.js

```diff
@@ -0,0 +1,251 @@
+
+    tx.moveCall({
+        target: `${gasServicePackageId}::gas_service::collect_gas`,
+        arguments: [
+            tx.object(gasServiceConfig.objects.gas_service),
+            tx.object(gasServiceConfig.objects.gas_collector_cap),
+            tx.pure.address(receiver), // Receiver address
+            tx.pure.u64(atomicAmount), // Amount
+        ],
+    });
+
+    const receipt = await broadcast(client, keypair, tx);
+
+    printInfo('Gas collected', receipt.digest);
+}
+
+async function refund(config, chain, args, options) {
+    const [keypair, client] = getWallet(chain, options);
+    const walletAddress = keypair.toSuiAddress();
+
+    const gasServiceConfig = chain.contracts.axelar_gas_service;
+    const gasServicePackageId = gasServiceConfig.address;
+    const gasServiceObjectId = gasServiceConfig.objects.gas_service;
+
+    const [messageId, amount] = args;
+    const receiver = options.receiver || walletAddress;
+
+    const atomicAmount = getAtomicAmount(amount);
+
+    const bytes = await getBcsBytesByObjectId(client, gasServiceObjectId);
+    const { balance: gasServiceBalance } = gasServiceStruct.parse(bytes);
+
+    // Check if the gas service balance is sufficient
+    if (gasServiceBalance < atomicAmount) {
+        printError('Insufficient gas service balance', `${getFormattedAmount(gasServiceBalance)} < ${getFormattedAmount(atomicAmount)}`);
+        return;
+    }
+
+    const tx = new TransactionBlock();
+    tx.moveCall({
+        target: `${gasServicePackageId}::gas_service::refund`,
+        arguments: [
+            tx.object(gasServiceConfig.objects.gas_service),
+            tx.object(gasServiceConfig.objects.gas_collector_cap),
+            tx.pure(bcs.string().serialize(messageId).toBytes()), // Message ID for the contract call
+            tx.pure.address(receiver), // Refund address
+            tx.pure.u64(atomicAmount), // Amount
+        ],
+    });
+
+    const receipt = await broadcast(client, keypair, tx);
+
+    printInfo('Gas refunded', receipt.digest);
+}
+
+async function processCommand(command, config, chain, args, options) {
+    const [keypair, client] = getWallet(chain, options);
+
+    await printWalletInfo(keypair, client, chain, options);
+
+    if (!chain.contracts.axelar_gas_service) {
+        throw new Error('Axelar gas service contract not found');
+    }
+
+    switch (command) {
+        case 'pay_gas':
+            printInfo('Action', 'Pay gas');
+            await payGas(config, chain, args, options);
+            break;
+        case 'add_gas':
+            printInfo('Action', 'Add gas');
+            await addGas(config, chain, args, options);
+            break;
+        case 'collect_gas':
+            printInfo('Action', 'Collect gas');
+            await collectGas(config, chain, args, options);
+            break;
+        case 'refund':
+            printInfo('Action', 'Refund gas');
+            await refund(config, chain, args, options);
+            break;
+    }
```

Comment:
the function itself should be provided as input to mainProcessor to avoid this switch case